File: backend/src/main/java/com/inklink/backend/Service/PortfolioItemService.java
package com.inklink.backend.service;
import com.inklink.backend.repository.PortfolioItemRepository;
import org.springframework.stereotype.Service;
import com.inklink.backend.model.PortfolioItem;
import com.inklink.backend.model.User;
import java.util.List;

@Service
public class PortfolioItemService
{
    private final PortfolioItemRepository repository;

    public PortfolioItemService(PortfolioItemRepository repository) {this.repository = repository;}

    public PortfolioItem addPortfolioItem(User owner, String fileUrl, String title, String description)
    {
        PortfolioItem item = new PortfolioItem();
        item.setOwner(owner);
        item.setFileURL(fileUrl);
        item.setTitle(title);
        item.setDescription(description);
        return repository.save(item);
    }

    public List<PortfolioItem> getPortfolioByOwner(Long ownerId) {
        return repository.findByOwnerId(ownerId);
    }
}

File: frontend/src/app/services/portfolio-item.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { PortfolioItem } from '../models/portfolio-item.model';

@Injectable({providedIn: 'root'})
export class PortfolioItemService {

  private apiUrl = 'http://localhost:8080/api/portfolioItems';

  constructor(private http: HttpClient) { }

  // invia il file vero tramite FormData (multipart/form-data) invece di un
  // semplice URL testuale; il backend lo salva su disco e restituisce
  // l'oggetto creato con il percorso dell'immagine appena salvata
  addPortfolioItem(file: File, title: string, description: string): Observable<PortfolioItem> {
    const formData = new FormData();
    formData.append('file', file);
    formData.append('title', title);
    formData.append('description', description);
    return this.http.post<PortfolioItem>(`${this.apiUrl}/upload`, formData);
  }

  getPortfolioByOwner(ownerId: number): Observable<PortfolioItem[]>{
    return this.http.get<PortfolioItem[]>(`${this.apiUrl}/owner/${ownerId}`);
  }
}
